### PR TITLE
fix warnings&errors of doc building

### PR DESCRIPTION
**Pull request by kudep**
# Description

*Please describe here what changes are made and why. If this pull request fixes a GitHub issue, reference it.*

# Checklist

- [x] I have covered the code with tests
- [x] I have added comments to my code to help others understand it
- [x] I have updated the documentation to reflect the changes
- [x] I have performed a self-review of the changes

### CODE REVIEW

**Review by pseusys — APPROVED** [review]

**Review by pseusys — APPROVED** [review]

File: dff/core/engine/conditions.py [hunk 1]
```diff
@@ -21,13 +21,14 @@
 
 @validate_arguments
 def exact_match(match: Any, *args, **kwargs) -> Callable:
```
Comment (pseusys) [inline dff/core/engine/conditions.py:23]: Maybe we should specify `Callable` signature?
It should be something like `Callable[[Context, Actor, Any, Any], bool]` (AFAIK) here.

File: dff/core/engine/conditions.py [hunk 2]
```diff
@@ -39,18 +40,15 @@ def exact_match_condition_handler(ctx: Context, actor: Actor, *args, **kwargs) -
 
 @validate_arguments
 def regexp(pattern: Union[str, Pattern], flags: Union[int, re.RegexFlag] = 0, *args, **kwargs) -> Callable:
```
Comment (pseusys) [inline dff/core/engine/conditions.py:42]: Should we add Callable signature (☝️): `Callable[[Context, Actor, Any, Any], bool]` (AFAIK) here.

File: dff/core/engine/conditions.py [hunk 3]
```diff
@@ -79,23 +77,25 @@ def check_cond_seq(cond_seq: list):
...
+"""
 _all = all
-""" _all is an alias for all. """
+"""
+_all is an alias for all.
+"""
 
 
 @validate_arguments
 def aggregate(cond_seq: list, aggregate_func: Callable = _any, *args, **kwargs) -> Callable:
```
Comment (pseusys) [inline dff/core/engine/conditions.py:90]: Should we add Callable signature (☝️): `Callable[[Context, Actor, Any, Any], bool]` (AFAIK) here.

File: dff/core/engine/conditions.py [hunk 4]
```diff
@@ -111,12 +111,13 @@ def aggregate_condition_handler(ctx: Context, actor: Actor, *args, **kwargs) ->
 
 @validate_arguments
 def any(cond_seq: list, *args, **kwargs) -> Callable:
```
Comment (pseusys) [inline dff/core/engine/conditions.py:113]: Should we add Callable signature (☝️): `Callable[[Context, Actor, Any, Any], bool]` (AFAIK) here.

File: dff/core/engine/conditions.py [hunk 5]
```diff
@@ -128,12 +129,13 @@ def any_condition_handler(ctx: Context, actor: Actor, *args, **kwargs) -> bool:
 
 @validate_arguments
 def all(cond_seq: list, *args, **kwargs) -> Callable:
```
Comment (pseusys) [inline dff/core/engine/conditions.py:131]: Should we add Callable signature (☝️): `Callable[[Context, Actor, Any, Any], bool]` (AFAIK) here.

File: dff/core/engine/conditions.py [hunk 6]
```diff
@@ -145,13 +147,13 @@ def all_condition_handler(ctx: Context, actor: Actor, *args, **kwargs) -> bool:
 
 @validate_arguments
 def negation(condition: Callable, *args, **kwargs) -> Callable:
```
Comment (pseusys) [inline dff/core/engine/conditions.py:149]: Should we add Callable signature (☝️): `Callable[[Context, Actor, Any, Any], bool]` (AFAIK) here.

File: dff/core/engine/conditions.py [hunk 7]
```diff
@@ -169,21 +171,18 @@ def has_last_labels(
     **kwargs,
 ) -> Callable:
```
Comment (pseusys) [inline dff/core/engine/conditions.py:172]: Should we add Callable signature (☝️): `Callable[[Context, Actor, Any, Any], bool]` (AFAIK) here.

File: dff/core/engine/conditions.py [hunk 8]
```diff
@@ -201,7 +200,11 @@ def has_last_labels_condition_handler(ctx: Context, actor: Actor, *args, **kwarg
 
 @validate_arguments
 def true(*args, **kwargs) -> Callable:
```
Comment (pseusys) [inline dff/core/engine/conditions.py:202]: Should we add Callable signature (☝️): `Callable[[Context, Actor, Any, Any], bool]` (AFAIK) here.

File: dff/core/engine/conditions.py [hunk 9]
```diff
@@ -211,7 +214,11 @@ def true_handler(ctx: Context, actor: Actor, *args, **kwargs) -> bool:
 
 @validate_arguments
 def false(*args, **kwargs) -> Callable:
```
Comment (pseusys) [inline dff/core/engine/conditions.py:216]: Should we add Callable signature (☝️): `Callable[[Context, Actor, Any, Any], bool]` (AFAIK) here.

**Review by pseusys — APPROVED** [review]
Docs look enjoyable!

**Comment by pseusys** [comment]
I would rather copy return type definition to each function definition as well

**Comment by avsakharov** [comment]
> I would rather copy return type definition to each function definition as well

Yes, that's better.

File: dff/core/pipeline/conditions.py [hunk 10]
```diff
@@ -13,18 +13,23 @@
 def always_start_condition(_: Context, __: Actor) -> bool:
     """
     Condition that always allows service execution, it's the default condition for all services.
-    :ctx: - current dialog context.
-    :actor: - pipeline actor.
     Returns bool (True).
+
+    :param ctx: current dialog context.
+    :type ctx: Context
```
Comment (kudep) [inline dff/core/pipeline/conditions.py:19]: Do we need :type somesting: yet?

Comment (avsakharov) [reply dff/core/pipeline/conditions.py:19]: No. I think I forgot to delete this line.

**Pull request merged**